crates/volta-core/src/distro/node.rs

```diff
@@ -41,6 +40,7 @@ cfg_if::cfg_if! {
 /// A provisioned Node distribution.
 pub struct NodeDistro {
     archive: Box<dyn Archive>,
+    staging_file: Option<NamedTempFile>,
```

Comment:
A thought on seeing this and how it's used—I'd be inclined to have two separate types here: one for the staging version and one for the version where it has successfully downloaded. That way, it's always transparently obvious from the type signatures alone whether you're dealing with a version that is in-progress or one that is in fact ready to be used. Then we could explicitly transform from the one to the other.

I tend to lean toward that pattern over using an `Option` many times, as it makes it possible to use the types to express the legal flows through the program. However, it's possible the refactor that would do that is larger than we want to take on, or that the flow here with `Option` is clear *enough*, so this is by no means a blocker—your call!

Reply:
I agree, and I initially considered using an Enum as the different types, but I like the idea of having distinct types that we can transform (we do this already with `PlatformSpec` and `Image`). I think this fits into the larger refactor I'm doing around the entire fetching process, which will likely get rid of the `Distro` trait entirely, because its abstraction breaks down horribly for packages.

In the implementation for that, however, I'll be sure to use distinct types to represent the different cases, instead of an `Option` :+1:

Reply:
Sweet; I look forward to seeing it!